firefox-ios/Ecosia/Analytics/Analytics.swift

```diff
@@ -254,6 +255,23 @@ open class Analytics {
             .label(label?.rawValue))
     }
 
+    // MARK: In-App Search
+    public func inappSearch(url: URL) {
+        guard NativeSRPVAnalyticsExperiment.isEnabled,
+              let query = url.getEcosiaSearchQuery() else {
+            return
+        }
+        let payload: [String: Any?] = [
+            "query": query,
+            "page_num": url.getEcosiaSearchPage(),
+            "plt_name": "ios",
+            "plt_v": Bundle.version as NSObject,
+            "search_type": url.getEcosiaSearchVerticalPath()
+        ]
+        track(SelfDescribing(schema: Self.inappSearchSchema,
```

Comment:
What happens if we send fewer of such values (e.g. in case we'd get `nil` out of `url.getEcosiaSearchPage()`)?

Reply:
The [schema we created](https://console.snowplowanalytics.com/308c7b08-d7d7-424b-97cc-2ad1212a6c90/data-structures/d36d4acdd88edbe9c7810b4ac433273fe62bb28f52a4f69382f86774e09d610d) only has query as required, the others are optional and will just not be included if `nil`. That's also why I added the `compactMapValues`.